### PR TITLE
Update Ruby Chess Instructions to add reference to popular notation systems

### PR DESCRIPTION
**Pull request by crespire**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**
As someone that didn't play Chess, but was highly engaged in the Discord, I learned about the PGN and FEN systems. Some learners may not play much Chess and not visit the discord, and might benefit from having been directed to these popular and standard game notation systems that allow for interoperability with popular chess sites and other chess programs.

**2. This PR:**
Adds a reference to PGN and FEN to direct those that might not play Chess at all become aware of these systems of notation.

**3. Additional Information:**
https://discord.com/channels/505093832157691914/690591236922409012/988834238747000892

### CODE REVIEW

**Review by dm-murphy — COMMENTED** [review]
Hi @crespire 👋 

Thanks for bringing this one up and making the PR. In the Discord conversation you linked, you mentioned adding this note into the Additional Resources section. But in the PR you put it in the Assignment.

What are your thoughts here? Do you think we should review adding PGN and FEN as part of the assignment requirements or as something that would be additional/optional to the assignment?

**Comment by crespire** [comment]
> Hi @crespire 👋
> 
> Thanks for bringing this one up and making the PR. In the Discord conversation you linked, you mentioned adding this note into the Additional Resources section. But in the PR you put it in the Assignment.
> 
> What are your thoughts here? Do you think we should review adding PGN and FEN as part of the assignment requirements or as something that would be additional/optional to the assignment?

Thanks for the prompt, Daniel! You're right, I originally thought we should include it as part of the additional resources before looking over the lesson again.

Having taken a look over the content, I ended up adding it to the serialization portion of the project prompt as PGN or FEN are ideal candidates to help people implement that feature. It allows folks to save/load a variety of boards and games from any source (like chess.com or lichess.com, for example).

Happy to adjust and stick it in additional resources, but they are specifically marked as "optional" which I don't think conveys properly the usefulness of having your program generate/accept PGN or FEN notation.

**Comment by ChargrilledChook** [comment]
I guess for me part of the beauty of this project is how little information it gives you, and how little guidance it provides. Realizing you can create something complex without any aides is part of what makes completing it so satisfying. 

I think part of the problem solving aspect of it is researching the domain, where you'll likely come across these notation concepts. 

I'm not opposed to adding some cues in but I'd probably prefer to keep the main instructions as agnostic as possible.

Open to other viewpoints!

**Comment by dm-murphy** [comment]
Thanks for sharing your thoughts here @ChargrilledChook and I was thinking along the same lines. One of the most valuable aspects of the chess project is that the assignment instructions don't walk you through what to do or how to do it. Essentially, this is chess, now go build it.

I appreciate your response as well @crespire and hearing more about your reasoning between adding this to Assignments vs. Additional Resources.

We're going to share this PR with the rest of the Ruby maintainer team first and then decide from there. Please bear with us, thanks! 🙂

**Comment by crespire** [comment]
I think my view here is a little biased from my own experience having never played Chess at all before encountering this project. I also see implementing FEN in my own project as a foundational element that allowed me to complete the entire project successfully due to the ease of testing scenarios.

I totally understand not wanting to give too much away. I do think leaving some sort of breadcrumb about standard notation systems would be helpful, so perhaps there might be some way to accomplish this while also keeping some of the mystery and self-driven research elements you guys have flagged.

**Comment by JoshDevHub** [comment]
Was doing some digging on this, and came across this [discussion in the discord](https://discord.com/channels/505093832157691914/540903304046182425/927221223824457728) about it. Just linking to show more opinions.

My personal opinion is that I think it should appear somewhere on the page (possibly in Additional Resources). I'm an experienced chess player, and I knew going in that FEN was something I'd want to use. And although it's not super obscure or anything, I do think a typical chess novice may have to do quite a bit of research before coming across it independently.

That said, I think clues about how it can be used or the problems it can help with (serialization, testing, 3 or 5 fold repetitions, etc.) should be kept absent. I agree that part of the beauty of the project is how hands-off it is.

**Comment by dm-murphy** [comment]
Thanks for the input @JoshDevHub 🙂 

@crespire The Ruby maintainers have discussed this and we would like to include a link about chess notation in the Additional Resources section. We want it to be something that learners could explore without being directly told to check out FEN or PGN.

The best resource example we have for now is the Chess Notation wiki page that includes notation systems for humans (e.g. algebraic notation) and notation systems for computers (e.g. FEN, PGN): https://en.wikipedia.org/wiki/Chess_notation 

But we would also be open to a different resource if there might be a better option out there.

Additionally, we are open to ideas about making some reference to chess notation in the Assignment instructions too. Something that could encourage learners to check out the link in the Additional Resources section. But we want to avoid giving anything away or making any specific instructions on what notations to use or how to implement them. Ultimately we don't want learners to feel that they have to do it this way or have to use chess notation at all.

@crespire please let me know if you are up for making these changes. You can just make new commits to this same PR and we can take it from there. This might take a few iterations to get right but could end up helping other learners in the long run. Thanks! :-)

**Comment by crespire** [comment]
> Thanks for the input @JoshDevHub 🙂
> 
> @crespire The Ruby maintainers have discussed this and we would like to include a link about chess notation in the Additional Resources section. We want it to be something that learners could explore without being directly told to check out FEN or PGN.
> 
> The best resource example we have for now is the Chess Notation wiki page that includes notation systems for humans (e.g. algebraic notation) and notation systems for computers (e.g. FEN, PGN): https://en.wikipedia.org/wiki/Chess_notation
> 
> But we would also be open to a different resource if there might be a better option out there.
> 
> Additionally, we are open to ideas about making some reference to chess notation in the Assignment instructions too. Something that could encourage learners to check out the link in the Additional Resources section. But we want to avoid giving anything away or making any specific instructions on what notations to use or how to implement them. Ultimately we don't want learners to feel that they have to do it this way or have to use chess notation at all.
> 
> @crespire please let me know if you are up for making these changes. You can just make new commits to this same PR and we can take it from there. This might take a few iterations to get right but could end up helping other learners in the long run. Thanks! :-)

Very happy to take this one and submit a first draft!

File: ruby/conclusion/project_ruby_final.md
Comment (crespire) [inline ruby/conclusion/project_ruby_final.md:19]: I added this line as I feel it's a good, but vague prompt to check out the Additional Resources section. Here, there will be a like to the previously included "Illustrated rules of Chess" plus the new "Chess notation" link.

The prompt is short, and just gives folks a push to scroll down if they're a bit out of their element.

Comment (ChargrilledChook) [reply ruby/conclusion/project_ruby_final.md:19]: Great idea 💡

File: ruby/conclusion/project_ruby_final.md
Comment (ChargrilledChook) [inline ruby/conclusion/project_ruby_final.md:28]: I'd consider expanding the description here to give a bit more background / context, but not a blocker for me

**Review by ChargrilledChook — APPROVED** [review]
@crespire Simple but direct and gets the job done, I like it. I have one small piece of feedback but it's not a merge blocker for me. Thanks for doing this!

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [reply ruby/conclusion/project_ruby_final.md:19]: 👍 

I think this is a good first iteration and nothing in the Assignment instructions gives anything away. We can always expand on this if needed.

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [reply ruby/conclusion/project_ruby_final.md:28]: @ChargrilledChook Any specific suggestions on how you would want to see this expanded? 

I'm also good with it as is since I think the link text is very clear. 

But I could also see something like: "Chess notation for humans and computers"

That might give some idea for learners to think about the differences between notation for humans vs. computers and how they might utilize one or the other.

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:29]: **thought (non-blocking):** It might be time to reconsider our inclusion of the RubyQuiz hints or rewording the description. 

If this new PR specifically directs learners to check out the Additional Resources then we might want to be more clear that these RubyQuiz hints should be a last resort. We might want to strike them altogether. 

Non-blocking since this is really a separate issue but I will bring it up for discussion with the Ruby team.

**Review by dm-murphy — APPROVED** [review]
Thanks @crespire 🙏 

I've approved this PR but added some thoughts too that might be worth discussing. I will bring this up one more time with the rest of the Ruby maintainer team too.

Appreciate all your efforts with this one! :-)

File: ruby/conclusion/project_ruby_final.md
Comment (ChargrilledChook) [reply ruby/conclusion/project_ruby_final.md:28]: I like that :ok_hand: but it's a fairly minor detail either way

File: ruby/conclusion/project_ruby_final.md
Comment (crespire) [reply ruby/conclusion/project_ruby_final.md:29]: @dm-murphy @ChargrilledChook 

I had thought about this, but didn't want to make any additional changes in my initial commit.

Having reviewd this resource, it basically contains an answer. How would folks feel if we just flagged that in the link?

I would suggest the following:

```markdown
* [A discussion of this problem from RubyQuiz](http://rubyquiz.com/quiz35.html). This link contains a blueprint for how to complete this project. If you're feeling stuck or need a hint, you might want to visit the community discord for help instead of looking up the answer!```

I think it a) lets folks know that it does contain a blueprint, with the corollary that if you care about solving this problem, it's not a good resource to pursue and b) prompts folks that might be stuck to check out the discord for help instead.

File: ruby/conclusion/project_ruby_final.md
Comment (crespire) [reply ruby/conclusion/project_ruby_final.md:28]: > @ChargrilledChook Any specific suggestions on how you would want to see this expanded?
> 
> I'm also good with it as is since I think the link text is very clear.
> 
> But I could also see something like: "Chess notation for humans and computers"
> 
> That might give some idea for learners to think about the differences between notation for humans vs. computers and how they might utilize one or the other.

I like that a lot, and I will push a commit to update the text of the link to "Chess notation for humans and computers"

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [reply ruby/conclusion/project_ruby_final.md:29]: Nice, I really like the mention of visiting the community Discord for help! 👍 

One more thought on this one. I liked the original language of "You don't need these. Resist the temptation!" 

What about starting off the bullet point with something like "You won't need this. Resist the temptation! But here is [a discussion of this problem from RubyQuiz]..."

Feel free to play with all of this and commit what you think might be best. Thanks again for working through these iterations @crespire!

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:29]:
```suggestion
* If you're feeling stuck, you should visit the community Discord!
```
**typo:** "visit"

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:17]:
```suggestion
  4. Write tests for the important parts.  You don't need to TDD it (unless you want to), but be sure to use RSpec tests for anything that you find yourself typing into the command line repeatedly.
```
**chore:** Fixing numbers

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:18]:
```suggestion
  5. Do your best to keep your classes modular and clean and your methods doing only one thing each.  This is the largest program that you've written, so you'll definitely start to see the benefits of good organization (and testing) when you start running into bugs.
```
**chore:** Fixing numbers

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:19]:
```suggestion
  6. Unfamiliar with Chess? Check out some of the additional resources to help you get your bearings.
```
**chore: Fixing numbers**

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:20]:
```suggestion
  7. Have fun!  Check out the [unicode characters](http://en.wikipedia.org/wiki/Chess_symbols_in_Unicode) for a little spice for your gameboard.
```
**chore:** Fixing numbers

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:21]:
```suggestion
  8. (Optional extension) Build a very simple AI computer player (perhaps who does a random legal move)
```
**chore:** Fixing numbers

File: ruby/conclusion/project_ruby_final.md
Comment (dm-murphy) [inline ruby/conclusion/project_ruby_final.md:30]: **praise:** Nice job! 👍

**Review by dm-murphy — CHANGES_REQUESTED** [review]
Great work @crespire!

There was one typo I added a suggestion comment for that you can just commit. 

I also went ahead and fixed the number order for the Assignment bullets. It was nothing on your end (it was previously incorrect) and the site would still show the correct numbered order anyway, but still think it's worth fixing the raw code. You can also just commit all those suggested fixes I made and it will push them through.

**Review by dm-murphy — APPROVED** [review]
Beautiful! ❤️ 

Thanks again for working through this one and for such great communication throughout these iterations! Really hope to see more contributions from you @crespire :-)

**Pull request merged**
